backend/product-listing.go

```diff
@@ -90,14 +126,10 @@ func initProducts(path string) {
 }
 
 func renderShoppingCart(w http.ResponseWriter, r *http.Request, page Page) {
-	page.Render(w,
-		ShoppingCartItem{
-			Img:      html.UnescapeString(r.URL.Query().Get("img")),
-			Name:     r.URL.Query().Get("name"),
-			Price:    r.URL.Query().Get("price"),
-			Quantity: r.URL.Query().Get("quantity"),
-		},
-	)
+	clientId := r.FormValue("clientId")
```

Comment:
The shopping cart page is an AMP page so I don't think I can use `history.replaceState(null, null, location.pathname)`, right?